Fix hr

yoyo2k opened the pull request:
Câteva chestii în plus pt hr

yoyo2k committed:
remove datetime import as its not needed

yoyo2k committed:
hr_employee_view

- tabs to spaces
- added ssn to view (echiv CNP)
- moved emit_XX to identification_id (echiv CI nr)

yoyo2k committed:
hr_employee

compute first and last name

yoyo2k committed:
hr_employee

- validare CNP
- setare automata a genului si datei de nastere

yoyo2k committed:
hr_employee

fixes

feketemihai commented:
:+1:  Daca crezi poti sa aranjezi un pic si vizualizarile pentru coasigurati si persoane in intretinere..ca acolo am pus doar campurile si sa punem ceva grup sa para mai ok si mai intuitiv ce trebuie sa completeze...

yoyo2k commented:
Să  le trecem la tabul hr. Mai intuitiv de nume cnp şi relatie nu ştiu

feketemihai commented:
Ma refeream cand deschizi form-ul direct din meniu...nu are nici un label...group...

yoyo2k commented:
Sau facem tab separat ?

yoyo2k commented:
Ok. Deci tab separat sau grup/label in hr settings că nu prea are ce să caute în personal sau , şi mai bine, opinii?

feketemihai commented:
Normal sunt informatii personale...dar care pot influenta calculul salariului...deci cred ca mai bine tab separat si vizualizarile pentru coasigurati si in intretinere sa fie de tip "form" cate un grup separat si pe linii. coloane...

yoyo2k commented:
Şi cum să numim tab-ul?
PS: merită să avem 2 tabele pentru acelasi tip de informatii? În loc de 2 tabele, o coloană cu trei stări, coasigurat , în îngrijire şi ambele.

feketemihai commented:
Pentru tab eu zic sa le lasam deocamdata asa...sa vedem ce mai trebuie sa adaugam pentru salarizare, si sa facem atunci un tab specific configuratiei de salarizare pentru Romania...

Pentru mine e ok si asa...o coloana in care sa selectezi tipul persoanei, si modif functia ce calculeaza nr persoane in intretinere.

yoyo2k commented:
De salarizare m-am apucat (modul separat). Mai am de lucru putin la module şi de implementat view-urile.

O singură chestie mai e la hr.. pt zilele libere.. Nu mă lasă nici in ruptul capului să creez alocarea direct pe tag ci doar după ce salvez, la editare , atunci mă lasă. E şi un issue pt gh https://github.com/odoo/odoo/issues/4713 . 

Există vreo unealtă sa-mi exporte fără view în data xml?

feketemihai commented:
Era pe 7 (daca nu pe 6.1) base module record, care iti exporta tranzactiile care le faci sub forma unui modul...datele in xml...

yoyo2k committed:
hr_employee:

- setare automata si a locului nasterii
- validare cu intregi
- un singur model pentru pers in ingrijire/coasig
- validare cnp

modificarile aferente view-ului & security

yoyo2k commented:
era pe 6.1 (ce am gasit eu). ceva pt 8?

feketemihai commented:
Depinde ce vrei sa exporti? nu mai bine exporti in csv si eu facem script cu care creeam xml...

import os
rootdir='/home/openerp/Documents/data'
def doWhatYouWant(line):
    return line
new=open('/home/openerp/Documents/mari.xml', 'w')

f=open('/home/openerp/Documents/data/ORASE.CSV', 'r')
lines=f.readlines()
for line in lines:
    line = line.replace('"','')
    line = line.replace('\n','')
    city=line.split(",")
    newline="""         <record id="city_%s"  model="city.city">
            <field name="zipcode">%s</field>
            <field name="name">%s</field>
            <field model="res.country.state" name="state_id" eval="%s"/>
            <field model="res.country" name="country_id" eval="base_ro"/>
            <field name="descriere">%s</field>  
        </record> """ % (city[3],city[3],city[1],city[0],city[2])
    new.write(newline + '\n')
f.close()

Ceva de genul...

yoyo2k commented:
vroiam sa evit export in csv ca sa-l prelucrez manual in xml.. dar daca nu e alternativa..

feketemihai commented on l10n_ro_hr/hr_employee.py:
```diff
@@ -20,51 +20,151 @@
-                                 ('firstdegree', 'First degree relationship'),
-                                 ('secdegree', 'Second degree relationship')],
-                                string='Person relation', required=True)
+def validate_cnp(cnp):
```
Aici nu ar fi mai usor sa folosim din stdnum.ro.cnp is_valid(cnp)...si apoi sa extragem data nasterii si localitatea? Ce zici...

feketemihai commented on l10n_ro_hr/hr_employee.py:
```diff
@@ -20,51 +20,151 @@
+    @api.onchange('ssnid')
+    @api.constrains('ssnid')
+    def _validate_ssnid(self):
+        if not validate_cnp(self.ssnid):
```
Aici trebuie pus si daca e completat altfel cel putin la mine cand adaug o linie da eroare pentru ca e bool:

 if self.ssnid and not validate_cnp(self.ssnid):

yoyo2k replied:
@api.constrains('ssnid') nu face asta?

feketemihai replied:
Nu...constrain pune o constrangere...nu inseamna ca e neaparat completat...poate daca pui depends...si atunci cred ca trebuie compleatat campul..

Mihai Fekete
Sent from Samsung Mobile.

-------- Original message --------
From: Adrian Vasile
Date:03/02/2015 13:28 (GMT+02:00)
To: odoo-romania/l10n-romania
Cc: Mihai FEKETE
Subject: Re: [l10n-romania] Fix hr (#50)

In l10n_ro_hr/hr_employee.pyhttps://github.com/odoo-romania/l10n-romania/pull/50#discussion_r23997616:

> -class hr_employee_coinsured(models.Model):
> -    _name = 'hr.employee.coinsured'
> -    _description = "Employee co-insured persons"
> -    @api.one
> -    @api.onchange('ssnid')
> -    @api.constrains('ssnid')
> -    def _validate_ssnid(self):
> -        if not validate_cnp(self.ssnid):

@api.constrains('ssnid') nu face asta?

## 

Reply to this email directly or view it on GitHubhttps://github.com/odoo-romania/l10n-romania/pull/50/files#r23997616.

feketemihai commented on l10n_ro_hr/hr_employee.py:
```diff
@@ -20,51 +20,151 @@
     employee_id = fields.Many2one('hr.employee', 'Employee', required=True)
     name = fields.Char(
-        'Person care name', required=True, help='Person in care name')
+        'Name', required=True, help='Person in care name')
```
help='Person related name'

feketemihai commented on l10n_ro_hr/hr_employee.py:
```diff
@@ -20,51 +20,151 @@
+    @api.constrains('ssnid')
+    def _ssnid_birthday_gender(self):
+        if self.country_id and 'RO' in self.country_id.code.upper():
+            if not validate_cnp(self.ssnid):
```
aici la fel verificare daca e completat ssnid dar e mai usor sa le punem in primul if...ca sa fii sigur si pentru data nasterii, locul nasterii, sex ca exista ssnid...

feketemihai commented on l10n_ro_hr/hr_employee_view.xml:
```diff
@@ -1,66 +1,61 @@
-							</tree>
-						</field>
-					</group>
+                    <group string="Persons in care">
```
string='Persons Related'

yoyo2k committed:
cele menționate de @fetekemihai

yoyo2k committed:
adaugat nume/prenume la coasig/in ingrijire

yoyo2k committed:
constrangere pentru relatie si tip

feketemihai commented:
Acum am vb cu manager HR de la noi si zicea ca ar fi de preferat ca la angajati sa avem si partea cu studii finalizate, cursuri, calificari, data la care trebuie efectuat urmatorul examen medical (fisa aptitudini)..asta este (poate punem un cron inainte cu 5-10 zile de averizare)...si poate valabilitate CI (deocamdata avem doar seria+numar, emis de xxx la data de xxx).

Voi ce ziceti?

yoyo2k commented:
trebuie sa avertizezi tu salariatu ca-i expira buletinu?? in declaratii conteaza cnp-ul, nu?

feketemihai commented:
In unele adeverinte trebuie sa treci data la care expira...

yoyo2k committed:
expires_on

yoyo2k commented:
studiile se refera doar la cele scolare aka liceale, universitare, etc?
din cursuri nu rezulta calificari? participa la cursul de frecat menta si va rezulta ca e calificat sa frece menta.

feketemihai commented:
Da...hai sa luam la stivuitorist...trebuie sa aiba un curs calificat pentru asta...ca de studii...

yoyo2k commented:
Fisa de aptitudine are valabilitate un an si poate avea urmatoarele avize de specialitate: apt, apt conditionat, inapt temporar, inapt, si apreciaza starea de sanatate a angajatului in functie de locul de munca.

yoyo2k commented:
tocmai.. calificarea nu implica cursul? deci care ar fi sensul cursului?

feketemihai commented:
Da...dar la semnare act aditional daca se schimba incadrarea automat trebuie alta fisa de aptitudini...care e iar valabila un an...deci la copierea contractului ca act aditional sa avem asta in vedere...

feketemihai commented:
Atunci lasa doar studii si calificari...

yoyo2k commented:
pentru expirarea examenului medical nu mai bine facem un filtru? about to expire

feketemihai commented:
Din partea mea e mai comod sa primesti un email cu 5-10 zile inainte sa fii instiintat ca acei angajati trebuie sa fie programati pt examenul medical...

yoyo2k commented:
si studiile si calificarile.. one2many sau camp ?

feketemihai commented:
Cred ca e mai ok one2many: an, institutia(res.partner), name(diploma de care beneficiaza/calificare)...

yoyo2k commented:
si hr-istu defapt nu foloseste odoo? n-are rapoarte configurate? pentru mine asta e printre motivele pt care folosesc odoo (one 2 rule them all exc. mail)

feketemihai commented:
Depinde de hr-ist si cate firme are in portofoliu...daca ai 5-10 nu prea iti vine sa intrii pe fiecare sa faci filtru...cel putin asa mi se pare...

yoyo2k commented:
sa facem parteneri din toate institutiile de inv?

feketemihai commented:
Sa si-i creeze ei...si alea au cui...

yoyo2k commented:
pai hr-istu are multi-company instalat, isi face filtru de expirare si dupa grupare pe companie... nu e deloc greu si iesit din comun. cum am mai spus one 2 rule them all.

ok. ce zici de:
hr_employee_studies:
  name - diploma
  institute - partner
  qualification - calificarea
ex:
- bac 2050/ min ed. / muncitor necalificat
- absolvent fac / min. ed. / telecomunicatii/pedagogie (in ideea ca a facut cursu din facultate de pedagogie)
- diploma de participare / min ed / frecator de menta

feketemihai commented:
E ok...

yoyo2k committed:
calificari si studii

yoyo2k commented:
intrebare la persoane in intretinere: pot si eu si nevasta sa trecem copii in intretinere?

feketemihai commented:
Nu

Mihai Fekete
Sent from Samsung Mobile.

-------- Original message --------
From: Adrian Vasile
Date:03/02/2015 17:40 (GMT+02:00)
To: odoo-romania/l10n-romania
Cc: Mihai FEKETE
Subject: Re: [l10n-romania] Fix hr (#50)

intrebare la persoane in intretinere: pot si eu si nevasta sa trecem copii in intretinere?

## 

Reply to this email directly or view it on GitHubhttps://github.com/odoo-romania/l10n-romania/pull/50#issuecomment-72672265.

yoyo2k committed:
calcul automat al numarului de copii in functie de numarul de copii in ingrijire

feketemihai commented:
Daca e la unul celalalt nu mai declara

Mihai Fekete
Sent from Samsung Mobile.

-------- Original message --------
From: Adrian Vasile
Date:03/02/2015 17:40 (GMT+02:00)
To: odoo-romania/l10n-romania
Cc: Mihai FEKETE
Subject: Re: [l10n-romania] Fix hr (#50)

intrebare la persoane in intretinere: pot si eu si nevasta sa trecem copii in intretinere?

## 

Reply to this email directly or view it on GitHubhttps://github.com/odoo-romania/l10n-romania/pull/50#issuecomment-72672265.

yoyo2k commented:
De asemenea, de scos şi educatia.. Dacă nu m-am uitat înainte..https://github.com/OCA/hr/ hr_skill şi hr_employee_education 
ne îndepărtăm de scopul comunităţii (localizare)

yoyo2k commented:
A mai rămas atenționarea hr-istului De expirarea controlului medical. Ceva ala https://github.com/OCA/hr/tree/8.0/__unported__/hr_contract_state

feketemihai commented:
Da, ceva de genul pentru cron (dar trebuie  vazut daca iei si de acolo hr_contract_init, pentru acte aditionale daca merge...sau initialul in ce stadiu il pui)...partea cu educatia mi se pare destul de simpla in OCA (poate aici sa mai lucram).

Din punctul meu de vedere e mai ok sa folosim ce e in OCA, pentru viitor si pentru numarul de commiters si calitatea. in ultimul timp se mizeaza foarte mult...

Eu as muta si l10n-romania in OCA...dar trebuie vazut cat dintre noi vreau sa intre in OCA, taxele de 50 EUR nu cred ca sunt o problema (e o suma modica).

yoyo2k commented:
Nu cred că e o problema de a băga proiectul in OCA, problema e că nu văd de ce să lucrăm la ceva care există deja fără a avea legătura cu localizarea pe românia. A.. ca poate updatam traducerea.. poate.. O văd mai mult ca problema omului care face implementarea sistemului.

feketemihai commented:
Asta ziceam si eu...punem o bifa ca pentru a avea toate functionalitatile instalati din OCA/HR..urmatoarele module...si currency_rate_update ar trebui sa il scoatem de aici ca normal e doar o copie al celui de pe OCA/account-financial-tools...

yoyo2k commented:
Absolut, şi account_storno?

feketemihai commented:
Ideea e ca la fiecare commit cei ce nu au semnat acordul sunt instiintati...La storno(in 9 va fi inclus) si vat_on_payment(in 9 generat de cash accounting) deocamdata le lasam...pt vat_on_payment am vazut ca este ceva commit in account-payment, pentru storno din cate stiu nu este...este adaptat de3 pe o versiune de pe 6.0(daca bine imi amintesc), la fel si vat_on_payment...ramanand cu autorii de atunci...deci cred ca pana ce vor fi bine structurate in OCA, sa le lasam aici...

yoyo2k committed:
- eliminare studii & calificari (exista functionalitati echiv in OCA/hr) & setarea numarului de copii automat
- adaugat cautare pt expirarea examenului medical

yoyo2k committed:
alt layout pt conditii de munca

yoyo2k commented:
Am grupat conditiile de muncă în două coloane Şi am scos numele grupelor pt ca mi se păreau redundante.

feketemihai commented on l10n_ro_hr/hr_employee.py:
```diff
@@ -93,12 +225,16 @@ def _number_personcare(self):
-        'hr.employee.care', 'employee_id', 'Person in care')
-    person_coinsured_ids = fields.One2many(
-        'hr.employee.coinsured', 'employee_id', 'Coinsured Persons')
+    studies = fields.One2many(
```
Remove field....nu mai are referinta...

feketemihai commented on l10n_ro_hr/security/ir.model.access.csv:
```diff
@@ -1,8 +1,6 @@
-access_hr_employee_care_all,hr.employee.care,model_hr_employee_care,base.group_hr_manager,1,1,1,1
-access_hr_employee_coinsured,hr.employee.coinsured,model_hr_employee_coinsured,base.group_user,1,0,0,0
-access_hr_employee_coinsured_all,hr.employee.coinsured,model_hr_employee_coinsured,base.group_hr_manager,1,1,1,1
+access_hr_employee_related,hr.employee.care,model_hr_employee_care,base.group_user,1,0,0,0
```
hr_employee_related in loc de hr_employee_care...

yoyo2k committed:
more cleanup

hr_employee: scos studies
ir.model.access: inlocuit care -> related

yoyo2k commented:
e ok?

feketemihai commented on l10n_ro_hr/hr_employee.py:
```diff
@@ -21,50 +21,182 @@
-                                required=True)
+                                string='Relation', required=True)
+    relation_type = fields.Selection([('in_care', 'In Care'),
+                                      ('coinsured', 'Coninsured'),
```
typo Coninsured in Co insured

feketemihai commented:
un typo pt co insured..in rest e ok
:+1:

feketemihai commented on l10n_ro_hr/hr_contract_view.xml:
```diff
@@ -62,30 +62,26 @@
 				</field>
```
In Contract poti pune si invisible pentru date_end in functie de period_type...daca e pe perioada nedeterminata, adica bifat period_type, nu ar trebui sa afiseze date_end...
si poate period_type deasupra la date...

yoyo2k replied:
am incercat div attrs="{'invisible': [('suspended', '==', 'False')]}" dar nu vrea (nici cu 0)

feketemihai replied:
'False' fara '', in domeniu, si = in loc de ==...attrs="{'invisible': [('suspended', '=', False)]}", la fel si in perioada contract cu period_type in date_end...

feketemihai commented on l10n_ro_hr/hr_contract_view.xml:
```diff
@@ -62,30 +62,26 @@
+							<group col="2" name="col1">
+								<field name="pensioneer"/>
+								<field name="suspended"/>
+								<label for="sus_date_from" string="Duration"/>
```
string Duration Suspended... sa se inteleaga ca e pt suspendat

yoyo2k committed:
conform ultimelor disc

yoyo2k committed:
done

yoyo2k committed:
typo

yoyo2k commented:
merge? ca-i dau push de la mine ca am rezolvat conflictu

feketemihai commented:
da... :+1:

feketemihai commented:
Faci tu si pt vat_on_payment...

yoyo2k commented:
done & done
